tests/xharness/XmlResultParser.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+
+namespace xharness {
+	public static class XmlResultParser {
+
+		public enum Jargon {
+			TouchUnit,
+			NUnit,
+			xUnit,
+			Text,
+		}
+
+		// test if the file is valid xml, or at least, that can be read it.
+		public static bool IsValidXml (string path, out Jargon type)
+		{
+			type = Jargon.Text;
+			if (!File.Exists (path))
+				return false;
+
+			bool isXml = false;
+			using (var stream = File.OpenText (path)) {
+				string line;
+				while ((line = stream.ReadLine ()) != null) { // special case when get got the tcp connection
+					if (line.Contains ("ping"))
+						continue;
+					if (line.Contains ("TouchUnitTestRun")) {
+						type = Jargon.TouchUnit;
+						isXml = true;
+						break;
```

Comment:
why not return `true` here ? (and in other cases too)
less code, nothing (now or later) can't happen outside the `while` to affect the logic
no need for `isXml` local either
easier to debug a `false` case since a breakpoint to the last `return` will always be false.